code/game/atoms.dm

```diff
@@ -157,6 +159,9 @@
 	return
 
 /atom/proc/bullet_act(obj/item/projectile/P, def_zone)
+	if(SEND_SIGNAL(src, COMSIG_ATOM_BULLET_ACT, P, def_zone) & COMPONENT_CANCEL_ATTACK_CHAIN)
+		return

```

Comment:
This seems like a strange place to use that component given that bullet_act isn't really an attack chain.
Also unclear why you want to avoid calling `P.on_hit()` considering this will mean shooting an artifact with an ion rifle or such will just ignore the projectile.

Reply:
This allows something like, say, a signal-based shield to negate impacts.